zerver/lib/events.py

```diff
@@ -74,6 +72,14 @@ def get_raw_user_data(realm: Realm, client_gravatar: bool) -> Dict[int, Dict[str
             profiles_by_user_id[user_id][profile_field.field_id] = {
                 "value": profile_field.value
             }
+    return profiles_by_user_id
+
+
+def get_raw_user_data(realm: Realm, client_gravatar: bool, for_api: bool=False,
+                      include_custom_profile_fields: bool=True,
+                      include_delivery_emails: bool=False) -> Dict[int, Dict[str, str]]:
+    user_dicts = get_realm_user_dicts(realm.id)
+    profiles_by_user_id = get_custom_profile_field_values(realm.id)
```

Comment:
It seems like we could skip this query if `include_custom_profile_fields` is `False`.

Reply:
Yes, good catch! Thanks for pointing it out!